### PR TITLE
dollar sign macro checks. Dollar sign is only valid inside a cluster …

### PR DESCRIPTION
**Pull request by eam**
…definition

### CODE REVIEW

**Comment by eam** [comment]
@mrkev This is the behavior difference we stumbled across

**Comment by mrkev** [comment]
🙌 looks good

File: spec/expand/dollar_sign/dollar_operator.spec [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# outside of a cluster definition the dollar sign is not a metacharacter
+$foo
+$foo
```
Comment (xaviershay) [inline spec/expand/dollar_sign/dollar_operator.spec]: is there a good use case for this? Does it make more sense for this to be a parse error (or at least no results)?

Comment (eam) [inline spec/expand/dollar_sign/dollar_operator.spec]: Generally, non-metacharacters are literals. I don't *think* I'd oppose it being an error, as long as it still does the right thing inside cluster definitions. Mostly concerned about the syntax actually doing something it shouldn't -- right now it seems to be an alias for `@`.

**Comment by eam** [comment]
Added a few more tests to ensure the locality of `$` is correct.

File: spec/expand/dollar_sign/dollar_operator.spec (hunk 1 above)
Comment (xaviershay) [inline spec/expand/dollar_sign/dollar_operator.spec]: yeah ok definitely shouldn't act like `@`!

**Pull request merged**